src/ui/properties/SceneNodeEditor.js

```diff
@@ -23,90 +27,245 @@ const FogTypeOptions = [
+  const onChangeAvatarMaxDistance = useSetPropertySelected(editor, "avatarMaxDistance");
 
-  onChangeBackground = background => {
-    this.props.editor.setPropertySelected("background", background);
-  };
-
-  onChangeFogType = fogType => {
-    this.props.editor.setPropertySelected("fogType", fogType);
-  };
-
-  onChangeFogColor = fogColor => {
-    this.props.editor.setPropertySelected("fogColor", fogColor);
-  };
-
-  onChangeFogNearDistance = fogNearDistance => {
-    this.props.editor.setPropertySelected("fogNearDistance", fogNearDistance);
-  };
-
-  onChangeFogFarDistance = fogFarDistance => {
-    this.props.editor.setPropertySelected("fogFarDistance", fogFarDistance);
-  };
-
-  onChangeFogDensity = fogDensity => {
-    this.props.editor.setPropertySelected("fogDensity", fogDensity);
-  };
-
-  render() {
-    const node = this.props.node;
-
-    return (
-      <NodeEditor {...this.props} description={SceneNodeEditor.description}>
-        <InputGroup name="Background Color">
-          <ColorInput value={node.background} onChange={this.onChangeBackground} />
-        </InputGroup>
-        <InputGroup name="Fog Type">
-          <SelectInput options={FogTypeOptions} value={node.fogType} onChange={this.onChangeFogType} />
+  return (
+    <NodeEditor {...props} description={SceneNodeEditor.description}>
+      <InputGroup name="Background Color">
+        <ColorInput value={node.background} onChange={onChangeBackground} />
+      </InputGroup>
+      <InputGroup name="Fog Type">
+        <SelectInput options={FogTypeOptions} value={node.fogType} onChange={onChangeFogType} />
+      </InputGroup>
+      {node.fogType !== FogType.Disabled && (
+        <InputGroup name="Fog Color">
+          <ColorInput value={node.fogColor} onChange={onChangeFogColor} />
         </InputGroup>
-        {node.fogType !== FogType.Disabled && (
-          <InputGroup name="Fog Color">
-            <ColorInput value={node.fogColor} onChange={this.onChangeFogColor} />
+      )}
+      {node.fogType === FogType.Linear && (
+        <>
+          <NumericInputGroup
+            name="Fog Near Distance"
+            smallStep={0.1}
+            mediumStep={1}
+            largeStep={10}
+            min={0}
+            value={node.fogNearDistance}
+            onChange={onChangeFogNearDistance}
+          />
+          <NumericInputGroup
+            name="Fog Far Distance"
+            smallStep={1}
+            mediumStep={100}
+            largeStep={1000}
+            min={0}
+            value={node.fogFarDistance}
+            onChange={onChangeFogFarDistance}
+          />
+        </>
+      )}
+      {node.fogType === FogType.Exponential && (
+        <NumericInputGroup
+          name="Fog Density"
+          smallStep={0.01}
+          mediumStep={0.1}
+          largeStep={0.25}
+          min={0}
+          value={node.fogDensity}
+          onChange={onChangeFogDensity}
+        />
+      )}
+      <InputGroup name="Override Audio Settings">
```

Comment:
Perhaps add `info` with a link to the pannernode settings page on MDN?

Reply:
Info is a tooltip, so you can't actually click on links in it. Changing the UX of prop info seems out of scope for this PR. However, I agree that would be a nice thing to add.